serving/src/main/java/feast/serving/specs/CachedSpecService.java

```diff
@@ -191,21 +220,81 @@ public void scheduledPopulateCache() {
     return featureSets;
   }
 
+  /**
+   * Generate a feature to feature set mapping from the given feature sets map. Accounts for
+   * variations (missing project, feature_set) in string feature references generated by creating
+   * multiple entries in the returned mapping for each variation.
+   *
+   * @param featureSets map of feature set name to feature set specs
+   * @return mapping of string feature references to name of feature sets
+   */
   private Map<String, String> getFeatureToFeatureSetMapping(
       Map<String, FeatureSetSpec> featureSets) {
-    HashMap<String, String> mapping = new HashMap<>();
-
-    for (FeatureSetSpec featureSetSpec : featureSets.values()) {
-      for (FeatureSpec featureSpec : featureSetSpec.getFeaturesList()) {
-        FeatureReference featureRef =
-            FeatureReference.newBuilder()
-                .setProject(featureSetSpec.getProject())
-                .setName(featureSpec.getName())
-                .build();
-        mapping.put(
-            generateFeatureStringRef(featureRef), generateFeatureSetStringRef(featureSetSpec));
-      }
-    }
+    Map<String, String> mapping = new HashMap<>();
+
+    featureSets.values().stream()
```

Comment:
This means that how users are able to reference features varies from store to store, right?

Reply:
How so?